code/modules/projectiles/guns/energy.dm

```diff
@@ -89,13 +98,114 @@ GLOBAL_LIST_INIT(registered_cyborg_weapons, list())
 					return suit.cell
 	return null
 
+/obj/item/weapon/gun/energy/special_check(var/mob/user)
+
+	if(!..())
+		return
+	
+	if(hatch_open)
+		to_chat(user, "<span class='warning'>[src] has its cell cover still open!</span>")
+		return 0
+	return 1
+
+//Attempts to load A into src, depending on the type of thing being loaded and the load_method
+//reloading in a new energy cell
+/obj/item/weapon/gun/energy/proc/load_ammo(var/obj/item/A, mob/user)
+	if(self_recharge)
+		return
+	//only let's you load in power cells
+	if(istype(A, /obj/item/weapon/cell))
+		. = TRUE
+		var/obj/item/weapon/cell/AM = A
+
+		if(hatch_open)
+			if(power_supply)
+				to_chat(user, "<span class='warning'>[src] already has a power cell loaded.</span>") //already a magazine here
```

Comment:
magazine cells

Reply:
Changed the comment to be more clear it is for cells